src/unit/test_hashset.c

```diff
@@ -438,7 +437,13 @@ int test_safe_iterator(int argc, char **argv, int flags) {
     long num_returned = 0;
     hashsetIterator iter;
     hashsetInitSafeIterator(&iter, s);
-    while (hashsetNext(&iter, (void **)&j)) {
+
+    /* Using void* variable to avoid a cast that violates strict aliasing */
+    void *element;
+    while (hashsetNext(&iter, &element)) {
+        static_assert(sizeof(long) == sizeof(void *));
```

Comment:
Instead of `long`, we could use `intptr_t` throughout this test suite. It explicitly means pointer-sized integer.

Reply:
The only place where `sizeof(long) != sizeof(void *)` is windows. See https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models